golem-test-framework/src/components/k8s.rs

```diff
@@ -0,0 +1,441 @@
+// Copyright 2024 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::anyhow;
+use async_dropper::{AsyncDrop, AsyncDropper};
+use async_trait::async_trait;
+use k8s_openapi::api::core::v1::{Namespace, Pod, Service};
+use k8s_openapi::api::networking::v1::Ingress;
+use kube::api::{DeleteParams, PostParams};
+use kube::{Api, Client};
+use serde_json::json;
+use std::time::Duration;
+use tokio::process::{Child, Command};
+use tracing::{debug, error, info};
+use url::Url;
+
+#[derive(Debug, Clone)]
+pub struct K8sNamespace(pub String);
+
+impl Default for K8sNamespace {
+    fn default() -> Self {
+        K8sNamespace("default".to_string())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum K8sRoutingType {
+    Minikube,
+    Ingress,
+}
+
+pub type K8sPod = AsyncDropper<ManagedPod>;
+pub type K8sService = AsyncDropper<ManagedService>;
+pub type K8sRouting = AsyncDropper<ManagedRouting>;
+
+pub struct ManagedPod {
+    name: String,
+    namespace: K8sNamespace,
+}
+
+impl ManagedPod {
+    pub fn new<S: Into<String>>(name: S, namespace: &K8sNamespace) -> ManagedPod {
+        ManagedPod {
+            name: name.into(),
+            namespace: namespace.clone(),
+        }
+    }
+}
+
+#[async_trait]
+impl AsyncDrop for ManagedPod {
+    async fn async_drop(&mut self) {
+        info!("Stopping pod {}", self.name);
+
+        let name = self.name.clone();
+
+        let pods: Api<Pod> =
+            Api::namespaced(Client::try_default().await.unwrap(), &self.namespace.0);
+
+        let dp = DeleteParams {
+            grace_period_seconds: Some(0),
```

Comment:
:+1: 

This might be the reason of my issues with `Drop` (why I commented out this code).